pkg/kubectl/cmd/cp/cp.go

```diff
@@ -184,7 +184,7 @@ func (o *CopyOptions) Run(args []string) error {
 		return err
 	}
 
-	if len(srcSpec.PodName) != 0 && len(destSpec.PodName) != 0 {
+	if len(srcSpec.PodName) == 0 && len(destSpec.PodName) != 0 {
```

Comment:
I think this is a result of a _hack_ to allow colons in filenames, ref. https://github.com/kubernetes/kubernetes/commit/4a433102751d3a2ab6f84b5d65254f21fd4aac84. From what I understand, it is supposed to be `!=` and not `==`. The `==` (copying to pod) is handled in line 197. But yes, I do agree that the code more or less doesn't make that much sense.

In my PR https://github.com/kubernetes/kubernetes/pull/78622, I have fixed the handling of colons/":" in filenames (with some tests), and therefore refactored this code away. If both srcSpec and destSpec are pods, it should result in an error saying that pod to pod isn't supported.

Reply:
this PR resolve the issue when copy local file to a Pod , we need check whether the lcoal file exist. 
u can see https://github.com/kubernetes/kubernetes/issues/78879 for details.
After your PR is merged, I will re-edit this.

Reply:
Ahh, then it makes more sense @Pingan2017 ! I thought this was handled inside `copyToPod` (line 198), but it looks like that isn't the case. Nice catch! 😄 

I don't think we can be sure about the intention of this, since line 190 use `o.copyToPod(fileSpec{File: args[0]}, destSpec, &exec.ExecOptions{})`and not `o.copyToPod(srcSpec, destSpec, &exec.ExecOptions{})` (since the commit that introduced this wrote that it was to fix handling of ":" in the filenames.)

Would be awesome if you could write a tests to avoid regressions of this! 😄

Reply:
@Pingan2017 please update the code so that we have only a single `copyToPod` invocation, the rest is ok. Definitely worth syncing with @odinuge and include his fix for the case when both are specified with pod names.